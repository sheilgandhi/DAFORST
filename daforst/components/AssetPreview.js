import { View, Text, StyleSheet, Image, Pressable } from 'react-native';
import React from 'react';
import { COLORS } from '../colors';

const AssetPreview = ({ asset, navigation }) => {
  return (
    <Pressable
      style={styles.container}
      onPress={() =>
        navigation.push('AssetFull', {
          asset: asset,
        })
      }>
      <Image
        source={{ uri: `data:image/png;base64,${asset.imgUrl}` }}
        style={styles.image}
      />
      <View style={styles.information}>
        <Text style={styles.lines}>
          <Text style={styles.bold}>Name: </Text>
          {asset.name}
        </Text>
        <Text style={styles.lines}>
          <Text style={styles.bold}>Owner: </Text>
          {asset.owner}
        </Text>
        <Text style={styles.lines}>
          <Text style={styles.bold}>Description: </Text>
          {asset.description}
        </Text>
      </View>
    </Pressable>
  );
};

export default AssetPreview;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  image: {
    width: 100,
    height: 100,
  },
  information: {
    flex: 1,
    justifyContent: 'center',
    paddingLeft: 20,
    color: COLORS.black,
  },
  line: {
    marginVertical: 15,
  },
  bold: {
    fontWeight: 'bold',
  },
});
