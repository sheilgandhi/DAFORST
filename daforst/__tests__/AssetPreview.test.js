import React from 'react';
import renderer from 'react-test-renderer';
import AssetPreview from '../components/AssetPreview';

const navigation = {
  goBack: jest.fn(),
  navigate: jest.fn(),
  setOptions: jest.fn(),
};

const asset = {
  imgUrl: jest.fn(),
};

describe('AssetPreview', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<AssetPreview navigation={navigation} asset={asset} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
