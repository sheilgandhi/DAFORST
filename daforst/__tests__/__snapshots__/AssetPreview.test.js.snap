// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetPreview renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "data:image/png;base64,function () {
        return fn.apply(this, arguments);
      }",
      }
    }
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  />
  <View
    style={
      Object {
        "color": "#000",
        "flex": 1,
        "justifyContent": "center",
        "paddingLeft": 20,
      }
    }
  >
    <Text>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Name: 
      </Text>
    </Text>
    <Text>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Owner: 
      </Text>
    </Text>
    <Text>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Description: 
      </Text>
    </Text>
  </View>
</View>
`;
